# Attempting to Size the Psychiatry Specialty
For context: 

The AAMC https://www.aamc.org/data/workforce/reports/492556/1-1-chart.html lists  892,856 in total active specialty physicans and 38,205 (4.3%) psychiatrists for 2017.

Statista here https://www.statista.com/statistics/209424/us-number-of-active-physicians-by-specialty-area/ lists 525,439 total active "specialty" physicians in March of 2019 of which 54,935 (10.5%) are psychiatrists. 

Keeping in mind there's a shortage of psychiatrists in general: https://news.aamc.org/patient-care/article/addressing-escalating-psychiatrist-shortage/

Here, I'm going to try to scrape individuals from the NPI registry to see how many individuals are listed under the taxonomy of 'psychiatry'



In [2]:
import pandas as pd
import requests
import matplotlib
import numpy as np
import matplotlib.pyplot as plt
import re
import time
from bs4 import BeautifulSoup

In [65]:
#have to skip every 200 
skip=range(0,20000,200)

In [77]:
# #used this to try and see if there were more than 1200 
# skip=[1200,2400]

In [74]:
data=[]

In [75]:
#individual test
# link=f'https://npiregistry.cms.hhs.gov/api/?number=&enumeration_type=NPI-2&taxonomy_description=Psychiatry&first_name=&last_name=&organization_name=&address_purpose=&city=&state=&postal_code=&country_code=&limit=200&skip=300&version=2.0'
# response_npi = requests.get(link)
# data_npi = response_npi.json()

In [78]:
for num in skip:
    link=f'https://npiregistry.cms.hhs.gov/api/?number=&enumeration_type=NPI-2&taxonomy_description=Psychiatry&first_name=&last_name=&organization_name=&address_purpose=&city=&state=&postal_code=&country_code=&limit=200&skip={num}&version=2.1'
    response_npi = requests.get(link)
    data_npi = response_npi.json()
    results=data_npi['results']
    print('---------------------------------')
    print(results[0]['basic']['authorized_official_first_name'])
    print(len(results))
#     time.sleep(1)
    for result in results[:len(results)-1]:
        person=result['basic']
        print(result['basic']['authorized_official_first_name'])
        for address in result['addresses']:
            column_name='zip_'+ str(result['addresses'].index(address)+1)
            person[column_name]=address['postal_code']
#             data.append(person)

---------------------------------
CHUCK
200
CHUCK
DAVID
MARK
DAVID
ALBERTA
HATTI
MICKEY
NANCY
DARLENE
NEIL
MANUEL
MANOJ
RALPH
PAUL
SWAMPNA
GAIL
ANDREW
JUDE
JACQUELINE
BRONSON
LISA
MONICA
NORMAN
TROY
TALIN
MELISSA
STEVEN
BRYON
RANDOLPH
DEBORAH
JENNIFER
PENELOPE
BENNETT
FREDERICK
FREDERICK
FREDERICK
FREDERICK
MICHAEL
KEITH
MICHAEL
BENNETT
RICHARD
DANIEL
LEISA
STEVE
HITEN
ANTHONY
LUCKOSE
THOMAS
LAWRENCE
REBECCA
ANGELA
KAREN
FRANCES
EDWIN
LINDA
WILLIAM
HOWARD
FRANCISCO
PATRICK
FREDERICK
CRYSTAL
RICK
MAXWELL
D
D
D
ROBERT
KENNETH
DAVID
REBECCA
EDWARD
AMY
CHARLES
JOSEPH
ERIC
ROGER
LEFAWN
CHERYL
GREG
DAVIS
MELISSA
MELISSA
MELISSA
MELISSA
MELISSA
MELISSA
MELISSA
MELISSA
MELISSA
MELISSA
MELISSA
MELISSA
MELISSA
DAVE
WALTER
MICHAEL
CARL
LOUISE
DENNIS
JASON
OMAR
JORRI
MICHAEL
JEFFREY
WILLIAM
JILL
MELISSA
ANTHONY
JEROME
CRAIG
SUSAN
JASMINE
DARREL
CANDACE
ABDUL
ALEXANDRA
STEPHEN
ALBERTO
LARRY
ROYAL
FEBE
GEOFFREY
RUSSELL
RUTH
MONIQUE
SYAM
JOSH
CHRIS
IAN
DEBORAH
ROBIN
FREDERICK
NITIN
ANN MARE
TIMOTHY
R

In [70]:
df=pd.DataFrame(data)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 39800 entries, 0 to 39799
Data columns (total 20 columns):
authorized_official_credential             19834 non-null object
authorized_official_first_name             39800 non-null object
authorized_official_last_name              39800 non-null object
authorized_official_middle_name            19854 non-null object
authorized_official_name_prefix            18482 non-null object
authorized_official_name_suffix            1016 non-null object
authorized_official_telephone_number       39800 non-null object
authorized_official_title_or_position      39800 non-null object
deactivation_date                          2106 non-null object
enumeration_date                           39800 non-null object
last_updated                               39800 non-null object
name                                       39800 non-null object
organization_name                          39800 non-null object
organizational_subpart                     39800

In [71]:
df.duplicated().value_counts()

True     38606
False     1194
dtype: int64

In [79]:
df=df.drop_duplicates(keep='first')

In [81]:
df.shape

(1194, 20)

So, turns out there are only 1,194 Psychiatrists and Neurologists listed which is an estimated: 2-3% of the workforce. 

In [82]:
df.to_csv('npi_psychiatrists.csv',index=False)

In [4]:
names=pd.read_csv('/Users/lilyliu-krason/Documents/lede/foundations/data_studio_projects/psychiatrists_diversity/names/Names_2010Census.csv')

In [7]:
df=pd.read_csv('npi_psychiatrists.csv')

In [14]:
npi_names=pd.DataFrame(df.authorized_official_last_name.value_counts().reset_index().rename(columns={'index':'name','authorized_official_last_name':'count'}))

In [16]:
names.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 162254 entries, 0 to 162253
Data columns (total 11 columns):
name            162253 non-null object
rank            162254 non-null int64
count           162254 non-null int64
prop100k        162254 non-null float64
cum_prop100k    162254 non-null float64
pctwhite        162254 non-null object
pctblack        162254 non-null object
pctapi          162254 non-null object
pctaian         162254 non-null object
pct2prace       162254 non-null object
pcthispanic     162254 non-null object
dtypes: float64(2), int64(2), object(7)
memory usage: 13.6+ MB


In [17]:
merge=npi_names.merge(names,left_on='name',right_on='name',how='left')

In [18]:
merge

name  count_x      rank    count_y  prop100k  cum_prop100k  \
0          KING       19      34.0   465422.0    157.78      10347.92   
1         FALVO       17   26313.0      928.0      0.31      77763.40   
2         BROWN       15       4.0  1437026.0    487.16       2521.56   
3       MCGINTY       10    5697.0     6084.0      2.06      61994.83   
4         AILES        9   38902.0      567.0      0.19      80834.59   
5         SMITH        8       1.0  2442977.0    828.19        828.19   
6        CERISE        7   52619.0      394.0      0.13      83032.39   
7      FRIEDMAN        5     901.0    38374.0     13.01      39740.92   
8        FILTON        5       NaN        NaN       NaN           NaN   
9         ADAMS        5      42.0   427865.0    145.05      11534.72   
10      MANNING        4     434.0    74949.0     25.41      31335.29   
11     PHILLIPS        4      52.0   360802.0    122.31      12854.67   
12         VOIT        4   19885.0     1351.0      0.46      75325.22   
13       BENSON        4     365.0    86081.0     29.18      29460.45   
14     WILLIAMS        4       3.0  1625252.0    550.97       2034.39   
15       BAILEY        4      72.0   277845.0     94.19      14985.19   
16     THOMPSON        4      23.0   664644.0    225.32       8302.08   
17    SCHNEIDER        4     312.0   101290.0     34.34      27808.00   
18     ESPOSITO        4    1576.0    22789.0      7.73      46409.86   
19      JOHNSON        4       2.0  1932812.0    655.24       1483.42   
20   BIRMINGHAM        3    5297.0     6577.0      2.23      61132.42   
21       HOWARD        3      75.0   264826.0     89.78      15259.53   
22        NAGER        3   49824.0      421.0      0.14      82639.46   
23      RAMIREZ        3      28.0   557423.0    188.97       9325.07   
24        JONES        3       5.0  1425470.0    483.24       3004.80   
25    GUTIERREZ        3      67.0   293218.0     99.40      14507.15   
26        DAVIS        3       8.0  1116357.0    378.45       4172.31   
27       MARTIN        3      20.0   702625.0    238.19       7610.74   
28       DOOLEY        3    1527.0    23481.0      7.96      46024.91   
29        PATEL        3      95.0   229973.0     77.96      16932.03   
..          ...      ...       ...        ...       ...           ...   
922   PINKERTON        1    4195.0     8446.0      2.86      58340.52   
923     KILLIAN        1    2833.0    12717.0      4.31      53594.17   
924   PRAZENICA        1       NaN        NaN       NaN           NaN   
925     REWERTS        1   40313.0      543.0      0.18      81098.94   
926      ADRIAN        1    6201.0     5514.0      1.87      62982.45   
927      BEADLE        1    8277.0     4005.0      1.36      66263.77   
928     GARDNER        1     194.0   142894.0     48.44      22988.79   
929      SHADID        1   43511.0      496.0      0.17      81658.00   
930        HART        1     207.0   137184.0     46.51      23603.22   
931       WEEKS        1     831.0    41565.0     14.09      38792.35   
932      ERMANN        1  158432.0      102.0      0.03      89974.70   
933       IRFAN        1   32865.0      698.0      0.24      79547.19   
934     MOORMAN        1    4780.0     7399.0      2.51      59910.81   
935       EWING        1    1049.0    33314.0     11.29      41533.75   
936       WRENN        1    6566.0     5160.0      1.75      63640.66   
937     BELCHER        1    1243.0    28321.0      9.60      43555.09   
938      DOBBYN        1   67519.0      292.0      0.10      84752.60   
939   BERNSTEIN        1    2015.0    17916.0      6.07      49404.47   
940      BOELKE        1   62359.0      321.0      0.11      84211.58   
941     TRAMUTA        1  131379.0      129.0      0.04      88901.26   
942     HUSSAIN        1    2146.0    16909.0      5.73      50175.97   
943   RATCLIFFE        1    7674.0     4327.0      1.47      65411.33   
944      SALLAS        1   46138.0      462.0      0.16      82084.57